### ISSUE TITLE
❓ [feature request] tandoor-recipes SSL

### ISSUE THREAD

**Issue by blowk** [alexbelgium/hassio-addons#1013]
Is there a way to enable SSL?

**Comment by blowk** [comment]
I've tried to add the option, but testing and adjusting the code is needed.
https://github.com/alexbelgium/hassio-addons/pull/1020

**Comment by alexbelgium** [comment]
I've put it in the test repo https://github.com/alexbelgium/hassio-addons-test

**Comment by blowk** [comment]
I've managed to work this add-on with https (using Nging proxy manager), but only the images aren't loading because of mixed content.
It's seem when checking the console with `window.localStorage.BASE_PATH` it came op with http://mydomain.com (instead of https)

I came across this thread https://github.com/TandoorRecipes/recipes/issues/953.
I don't know if it's possible to change the nginx config for this Tandoor docker? Or maybe add/override your own nginx lines with 
`proxy_set_header X-Forwarded-Proto $scheme;` or `proxy_set_header set X-Forwarded-Proto "https"`
for example.

**Comment by alexbelgium** [comment]
Hi, that's great congrats! To modify the image the easiest way (that I often do) is to use this merholodogy : https://github.com/alexbelgium/hassio-addons/wiki/Add%E2%80%90ons-feature-:-customisation

It executes a custom script at boot, which for example would allow you to modify the nginx config (likely in /etc/nginx/servers/*.conf) by using sed to add the line you would want

**Comment by blowk** [comment]
Hi

Thanks for your help in advance.
I'm not familiar with custom script, but I will try.

Do I need to put the code in addons_config/tandoor-recipes/tandoor-recipes.sh, to adjust nginx config?

```
#!/bin/bash
echo "Starting script"

# Path to the Nginx configuration file
NGINX_CONF="/opt/recipes/nginx/conf.d"

# Check if the proxy_set_header line already exists
if grep -q "proxy_set_header set X-Forwarded-Proto" $NGINX_CONF; then
  echo "Modifying existing proxy_set_header line"
  # Modify the line
  sed -i "s|proxy_set_header set X-Forwarded-Proto \"https\"; # modified header|g" $NGINX_CONF
else
  echo "Adding new proxy_set_header line"
  # Add the line to the end of the file
  echo "proxy_set_header set X-Forwarded-Proto \"https\"; # modified header" >> $NGINX_CONF
fi

echo "... done"
```

**Comment by alexbelgium** [comment]
Hi, yes exactly! And when the add-on starts it should show your "echo" commands to show that the code was properly applied.

Another solution is to use portainer to enter the container and check that the file is indeed there. If you modify the nginx config from within the container you need to restart nginx with `nginx -s restart`. If you use the script injector as above, no need to restart as nginx is started at the end of the start sequence

**Comment by blowk** [comment]
Ok, thanks. Nice to hear that I'm on the right track.

Unfortunately, I had no luck.
If tried with the file _tandoor-recipes.sh_ like described, but it doesn't seems to be loaded.

```
/etc/cont-init.d/01-custom_script.sh: executing
Execute /homeassistant/addons_config/tandoor-recipes/tandoor-recipes.sh if existing
Wiki here : github.com/alexbelgium/hassio-addons/wiki/Add-ons-feature-:-customisation
/etc/cont-init.d/20-folders.sh: executing
/etc/cont-init.d/99-run.sh: executing
```
I have tried it with debug enabled en disabled, but I didnt show the echo command.

The script:
```
#!/bin/bash
echo "Starting script"
```

I also enter the container and changed the Recipes.conf. Only restarting with `nginx -s restart` didn"t work.

Do I need anything else to get it working?

**Comment by blowk** [comment]
**Update**:
I got the custom script working.
The log mentioned "/homeassistant/addons_config/tandoor-recipes/tandoor-recipes.sh if existing" but it was another folder to put the script in: "/homeassistant/addons_autoscripts/tandoor-recipes.sh"

**Comment by alexbelgium** [comment]
Glas that it works! I corrected that in the main script but didn't pushed a new tandoor add-on since

**Comment by blowk** [comment]
Hi @alexbelgium 
I got it working. Using nginx proxy in combination with cloudflare and now everthing is forced to use https.

**Comment by alexbelgium** [comment]
Glad to hear that it worked, congrats ! How did you do it : just using cloudflare to create a tunnel ?

**Comment by blowk** [comment]
Yes, using the Cloudflare integration https://www.home-assistant.io/integrations/cloudflare/.
and nginx to handle the portforwarding to Tandoor port and the right (sub)domain.
